codes/javascript/chapter_stack_and_queue/linkedlist_deque.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * File: linkedlist_deque.js
+ * Created Time: 2023-02-04
+ * Author: Zhuo Qinyue (1403450829@qq.com)
+ */
+
+class DequeNode {
+    prev;   // 前驱结点引用 (指针)
+    next;   // 后继结点引用 (指针)
+    val;    // 结点值
+    
+    constructor(val) {
+        this.val = val;
+        this.next = null;
+        this.prev = null;
+    }
+}
+
+/* 基于双向链表实现的双向队列 */
+class LinkedListDeque {
+    head;   // 头结点 head
+    tail;   // 尾结点 tail
+    len;    // 双向队列的长度
+
+    constructor() {
+        this.head = null;
+        this.tail = null;
+        this.len = 0;
+    }
+
+    /* 队尾入队操作 */
+    pushBack(val) {
+        const node = new DequeNode(val);
+        // 若链表为空，则令 head, tail 都指向 node
+        if (this.len === 0) {
+            this.head = node;
+            this.tail = node;
+        } else {
+            // 将 node 添加至链表尾部
+            this.tail.next = node;
+            node.prev = this.tail;
+            this.tail = node; // 更新尾结点
+        }
+        this.len++;
+    }
+
+    /* 队首入队操作 */
+    pushFront(val) {
```

Comment:
pushFirst

Please keep the code consistency with the reference code.

Reply:
Got it